agent/src/main/java/com/microsoft/applicationinsights/agent/internal/agent/sql/PreparedStatementClassDataProvider.java

```diff
@@ -230,8 +231,23 @@ public ClassVisitor create(ClassInstrumentationData classInstrumentationData, Cl
                 return new PreparedStatementClassVisitor(classInstrumentationData, classWriter, metaData1);
             }
         };
+    }
 
-        return classVisitorFactory;
+    private ClassVisitorFactory classFactoryForNewPostgreSql() {
+        return new ClassVisitorFactory() {
+            @Override
+            public ClassVisitor create(ClassInstrumentationData classInstrumentationData, ClassWriter classWriter) {
+                HashSet<String> ctorSignatures = new HashSet<String>();
+                ctorSignatures.add("(Lorg/postgresql/jdbc/PgConnection;Ljava/lang/String;III)V");
+                // NOTE: this constructor cannot be added as it does not have a String in position 2.
```

Comment:
If the instrumentation of this constructor is missing, what functionalities would be disabled?

Reply:
From what I can see this constructor is never directly called by clients as it is not a client responsibility to provide a CachedQuery (which is internal to the driver). The constructor which takes the sql statement (String) is the one clients call and that in turn calls the overloaded (CachedQuery) version of the constructor which deals with cached queries.  

https://github.com/pgjdbc/pgjdbc/blob/master/pgjdbc/src/main/java/org/postgresql/jdbc/PgPreparedStatement.java

I have removed that comment as it is not useful, actually it's probably confusing.